micrometer-test/src/main/java/io/micrometer/core/tck/MeterRegistryCompatibilityKit.java

```diff
@@ -472,6 +472,138 @@ void gaugesCannotBeRegisteredEffectivelyTwice() {
+        void timeGaugesCannotBeRegisteredTwice() {
+            TimeGauge.builder("my.tg", () -> 100, TimeUnit.MILLISECONDS).register(registry);
+            TimeGauge.builder("my.tg", () -> 200, TimeUnit.MILLISECONDS).register(registry);
+
+            assertThat(registry.get("my.tg").timeGauges()).hasSize(1);
+            assertThat(registry.get("my.tg").timeGauge().value(TimeUnit.MILLISECONDS)).isEqualTo(100);
+        }
+
+        @Test
+        @DisplayName("timeGauges cannot be registered effectively twice")
+        void timeGaugesCannotBeRegisteredEffectivelyTwice() {
+            registry.config().meterFilter(MeterFilter.ignoreTags("ignored"));
+            TimeGauge.builder("my.tg", () -> 300, TimeUnit.MILLISECONDS).tags("ignored", "3").register(registry);
+            TimeGauge.builder("my.tg", () -> 400, TimeUnit.MILLISECONDS).tags("ignored", "4").register(registry);
+
+            assertThat(registry.get("my.tg").timeGauges()).hasSize(1);
+            assertThat(registry.get("my.tg").timeGauge().value(TimeUnit.MILLISECONDS)).isEqualTo(300);
+        }
+
+    }
+
+    @DisplayName("multiGauges")
+    @Nested
+    class MultiGaugeTck {
+
+        @Test
+        @DisplayName("multiGauges can be registered twice because they take care about de-registering the previous Gauge if overwrite=true")
+        void multiGaugesCanBeRegisteredTwice() {
+            MultiGauge gauges = MultiGauge.builder("my.gauge").register(registry);
+            gauges.register(Collections.singletonList(MultiGauge.Row.of(Tags.of("test", "true"), 1)));
+
+            // no re-registration since overwrite is false by default
+            gauges.register(Collections.singletonList(MultiGauge.Row.of(Tags.of("test", "true"), 2)));
+            assertThat(registry.get("my.gauge").tags("test", "true").gauges()).hasSize(1);
+            assertThat(registry.get("my.gauge").tags("test", "true").gauge().value()).isEqualTo(1);
+
+            // re-registration since overwrite is set to true
+            gauges.register(Collections.singletonList(MultiGauge.Row.of(Tags.of("test", "true"), 3)), true);
+            assertThat(registry.get("my.gauge").tags("test", "true").gauges()).hasSize(1);
+            assertThat(registry.get("my.gauge").tags("test", "true").gauge().value()).isEqualTo(3);
+        }
+
+        @Test
+        @DisplayName("multiGauges can be registered effectively twice because they take care about de-registering the previous Gauge if overwrite=true")
+        void multiGaugesCanBeRegisteredEffectivelyTwice() {
+            registry.config().meterFilter(MeterFilter.ignoreTags("ignored"));
+            MultiGauge gauges = MultiGauge.builder("my.gauge").register(registry);
+            gauges.register(Collections.singletonList(MultiGauge.Row.of(Tags.of("ignored", "true"), 4)));
+
+            // no re-registration since overwrite is false by default
+            gauges.register(Collections.singletonList(MultiGauge.Row.of(Tags.of("ignored", "true"), 5)));
```

Comment:
It seems that the "ignored" tag has the same value with the above accidentally.

Changing it to have a different value makes the test fail.

`MultiGauge` re-registration doesn't seem to work with `MeterFilter.ignoreTags()`.

Reply:
I created https://github.com/micrometer-metrics/micrometer/pull/6146 to demonstrate it.

Reply:
Thank you for catching this. This test should assume that `MeterFilter` works on the `MultiGauge` too so I guess the same value in this test is ok but this is a bug we should fix.

Reply:
@jonatan-ivanov Thanks for the feedback!

While reading the other changes like the `FunctionCounterTck` one, I thought that the "effectively" means being eventually same meters by ignoring the "ignored" tag. That's why I expected different values for the ignored" tag here.

Reply:
I'm ok using different values here after the underlying bug is fixed.